optimum/onnxruntime/trainer.py

```diff
@@ -907,7 +907,7 @@ def evaluate(
         """
         # memory metrics - must set up as early as possible
         # TODO: We need to enable evaluation using ORT backend.
```

Comment:
in the PR description, it mentions "packages/optimum/onnxruntime/trainer_seq2seq.py", line 675, in prediction_step
    generation_inputs = inputs[self.model.main_input_name]"

but why we need fix trainer.py instead of trainer_seq2seq.py?

Reply:
we fixed trainer.py in optimum.
because that's where we have the ModuleWithLoss class.

this is the whole stack trace :
```
***** train metrics *****
  epoch                    =        1.0
  train_loss               =     0.0979
  train_runtime            = 0:08:14.83
  train_samples            =     610320
  train_samples_per_second =   1233.374
  train_steps_per_second   =       4.82
[INFO] Evaluating with PyTorch backend. If you want to use ONNX Runtime for the evaluation, set `trainer.evaluate(inference_with_ort=True)`.
'ModuleWithLoss' object has no attribute 'main_input_name'
Traceback (most recent call last):
  File "run_translation.py", line 680, in <module>
    main()
  File "run_translation.py", line 612, in main
    metrics = trainer.evaluate(
  File "/opt/conda/envs/ptca/lib/python3.8/site-packages/optimum/onnxruntime/trainer_seq2seq.py", line 92, in evaluate
    return super().evaluate(
  File "/opt/conda/envs/ptca/lib/python3.8/site-packages/optimum/onnxruntime/trainer.py", line 957, in evaluate
    output = eval_loop(
  File "/opt/conda/envs/ptca/lib/python3.8/site-packages/transformers/trainer.py", line 3213, in evaluation_loop
    loss, logits, labels = self.prediction_step(model, inputs, prediction_loss_only, ignore_keys=ignore_keys)
  File "/opt/conda/envs/ptca/lib/python3.8/site-packages/optimum/onnxruntime/trainer_seq2seq.py", line 675, in prediction_step
    generation_inputs = inputs[self.model.main_input_name]
  File "/opt/conda/envs/ptca/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1614, in __getattr__
    raise AttributeError("'{}' object has no attribute '{}'".format(
AttributeError: 'ModuleWithLoss' object has no attribute 'main_input_name'
0543863ffef24c09a57b3fc5ce0f584b000000:93:290 [0] NCCL INFO [Service thread] Connection closed by localRank 0
0543863ffef24c09a57b3fc5ce0f584b000000:93:93 [0] NCCL INFO comm 0x5b711860 rank 0 nranks 8 cudaDev 0 busId 100000 - Abort COMPLETE
ERROR:torch.distributed.elastic.multiprocessing.api:failed (exitcode: 1) local_rank: 0 (pid: 93) of binary: /opt/conda/envs/ptca/bin/python
Traceback (most recent call last):
  File "/opt/conda/envs/ptca/bin/torchrun", line 8, in <module>
    sys.exit(main())
  File "/opt/conda/envs/ptca/lib/python3.8/site-packages/torch/distributed/elastic/multiprocessing/errors/__init__.py", line 346, in wrapper
    return f(*args, **kwargs)
  File "/opt/conda/envs/ptca/lib/python3.8/site-packages/torch/distributed/run.py", line 794, in main
    run(args)
  File "/opt/conda/envs/ptca/lib/python3.8/site-packages/torch/distributed/run.py", line 785, in run
    elastic_launch(
  File "/opt/conda/envs/ptca/lib/python3.8/site-packages/torch/distributed/launcher/api.py", line 134, in __call__
    return launch_agent(self._config, self._entrypoint, list(args))
  File "/opt/conda/envs/ptca/lib/python3.8/site-packages/torch/distributed/launcher/api.py", line 250, in launch_agent
    raise ChildFailedError(
torch.distributed.elastic.multiprocessing.errors.ChildFailedError: 
```